src/main/java/com/example/onlineservicesemulator/Controller.java

```diff
@@ -3,43 +3,75 @@
 import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
 import javafx.fxml.Initializable;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
 import javafx.scene.control.CheckBox;
 import javafx.scene.control.Label;
+import javafx.scene.control.ListView;
+import javafx.scene.input.MouseEvent;
 import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
 
 import java.net.URL;
-import java.util.List;
-import java.util.ResourceBundle;
+import java.util.*;
 
 public class Controller implements Initializable {
 
+    @FXML
+    private ListView<String> filesList;
     @FXML
     private VBox checkboxList;
-
+    @FXML
+    private Button addFileButton;
+    private List<String> servicesNames;
+    private Map<String, List<String>> servicesAndUploadedFilesMap;
     @Override
     public void initialize(URL url, ResourceBundle resourceBundle) {
+        servicesNames = JSONReader.getServices();
         createAndSetCheckboxList();
+        setFilesMap();
+        addFileButton.setVisible(false);
     }
 
     public void createAndSetCheckboxList() {
-        List<String> servicesNames = JSONReader.getServices();
-        double lenghtSectionA = calculateLongestServiceString(servicesNames);
-        checkboxList.setPrefWidth(lenghtSectionA);
+        double lengthSectionA = calculateLongestServiceString(servicesNames)*8;
+        checkboxList.setPrefWidth(lengthSectionA);
         for (String serviceName : servicesNames) {
             CheckBox checkBox = new CheckBox();
             checkBox.setId("checkbox_" + serviceName);
             Label checkBoxLabel = new Label(serviceName);
             checkBox.setGraphic(checkBoxLabel);
-            checkBoxLabel.setOnMouseClicked(event -> {
-                checkBox.setSelected(false);
-                event.consume();
-            });
+            checkBoxLabel.setOnMouseClicked(event -> onCheckboxLabelClicked(event, checkBox, checkBoxLabel, serviceName));
             checkboxList.getChildren().add(checkBox);
             checkboxList.setSpacing(5);
         }
     }
 
+    private void onCheckboxLabelClicked(MouseEvent event, CheckBox checkBox, Label checkBoxLabel,String serviceName) {
+        checkBox.setSelected(false);
+        filesList.getItems().clear();
+        String clickedService = checkBoxLabel.getText();
+        if (servicesAndUploadedFilesMap.containsKey(clickedService) && servicesAndUploadedFilesMap.get(clickedService) != null) {
+            filesList.getItems().addAll(servicesAndUploadedFilesMap.get(clickedService));
+            addFileButton.setVisible(true);
+        }
+        event.consume();
+        addFiles(serviceName);
+    }
+
+    public void setFilesMap() {
+        servicesAndUploadedFilesMap = new HashMap<>();
+        servicesNames.forEach(service -> servicesAndUploadedFilesMap.put(service, null));
+
+        List<String> arraylist1 = new ArrayList<>();
+        arraylist1.add("windowFile1.dat");
```

Comment:
why is this hardcoded here, i pretty much have the same comments to your pull request as to radu''s, look at his for the things to change